modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/row/RowSchema.java

```diff
@@ -101,4 +100,35 @@ public RowSchema build() {
             return new RowSchema(types);
         }
     }
+
+    /**
+     * Creates a new schema based on original schema and provided mapping.
+     */
+    public static RowSchema map(RowSchema schema, int[] mapping) {
```

Comment:
Should `ModifyNode` also use this method instead of `getMappedRowSchema`?

Reply:
sorry, missed comment from @ygerzhedovich about the same thing

Reply:
Null or empty mapping make no sense for this method.
Why we should decide here, what null\empty mapping actually means?